src/rolling/directory.rs

```diff
@@ -140,7 +140,7 @@ impl RollingReader {
     ///
     /// If no block was read, positions itself at the beginning.
     pub async fn into_writer(mut self) -> io::Result<RollingWriter> {
-        let offset = self.block_id * crate::BLOCK_NUM_BYTES;
+        let offset = (self.block_id - 1) * crate::BLOCK_NUM_BYTES;
```

Comment:
fix bug where writer would start writing on the next block (but at the right offset), then making the reader believe it has reached end of data when there are still things in the next block